src/tsserver/nodeServer.ts

```diff
@@ -668,3 +661,60 @@ function startNodeSession(options: StartSessionOptions, logger: ts.server.Logger
         return combinePaths(normalizeSlashes(homePath), cacheFolder);
     }
 }
+
+function pipeExists(name: string): boolean {
+    // Unlike statSync, existsSync doesn't throw an exception if the target doesn't exist.
+    // A comment in the node code suggests they're stuck with that decision for back compat
+    // (https://github.com/nodejs/node/blob/9da241b600182a9ff400f6efc24f11a6303c27f7/lib/fs.js#L222).
+    // Caveat: If a named pipe does exist, the first call to existsSync will return true, as for
+    // statSync.  Subsequent calls will return false, whereas statSync would throw an exception
+    // indicating that the pipe was busy.  The difference is immaterial, since our statSync
+    // implementation returned false from its catch block.
+    return fs.existsSync(name);
+}
+
+function createCancellationToken(args: string[]): ts.server.ServerCancellationToken {
+    let cancellationPipeName: string | undefined;
+    for (let i = 0; i < args.length - 1; i++) {
+        if (args[i] === "--cancellationPipeName") {
+            cancellationPipeName = args[i + 1];
+            break;
+        }
+    }

```

Comment:
@jakebailey Perhaps `ts.server.findArgument()` can be used here as well?

https://github.com/microsoft/TypeScript/blob/30979c2651e5b0aa8ab583bbd8161cac3d0ea317/src/tsserver/nodeServer.ts#L94

Reply:
https://github.com/microsoft/TypeScript/pull/60436

Reply:
That was quick! Thanks.